pyhive/hive.py

```diff
@@ -252,6 +260,55 @@ def _fetch_more(self):
         for row in response.results.rows:
             self._data.append([_unwrap_col_val(val) for val in row.colVals])
 
+    def poll(self):
+        """Poll for and return the raw status data provided by the Hive Thrift REST API.
+        :returns: ``ttypes.TOperationState``
+        :raises: ``ProgrammingError`` when no query has been started
+        .. note::
+            This is not a part of DB-API.
+        """
+        if self._state == self._STATE_NONE:
+            raise ProgrammingError("No query yet")
+        
+        req = ttypes.TGetOperationStatusReq(
+            operationHandle=self._operationHandle
+        )
+        response = self._connection.client.GetOperationStatus(req)
+        _check_status(response)
+        
+        return response.operationState
```

Comment:
It looks like there's other useful attributes, so can you return the whole response?
https://github.com/apache/hive/blob/b6218275b00b64aed7efaf470784cc0441464f67/service-rpc/if/TCLIService.thrift#L1021